ttnn/ttnn/operations/core.py

```diff
@@ -946,4 +946,78 @@ def impl(file_name, tensor):
     ttl.tensor.decorate_external_operation(impl, function_name="ttnn.dump_tensor")(file_name, tensor)
 
 
+def _as_tensor_validate_input_tensors(operation_name, tensor, *args, **kwargs):
+    import torch
+
+    ranks = (1, 2, 3, 4, 5, 6, 7, 8)
+    if len(tensor.shape) not in ranks:
+        raise RuntimeError(f"{operation_name}: ttnn.Tensor must be of rank {ranks}, but got {len(tensor.shape)}")
+    dtypes = (torch.bfloat16, torch.float32, torch.int16, torch.int32, torch.int64, torch.float16)
+    if tensor.dtype not in dtypes:
+        raise RuntimeError(f"{operation_name}: ttnn.Tensor must be of type {dtypes}, but got {tensor.dtype}")
+    # if not tensor.is_contiguous():
+    #     raise RuntimeError(f"{operation_name}: ttnn.Tensor must be contiguous")
+
+
+@ttnn.register_operation(name="ttnn.as_tensor", validate_input_tensors=_as_tensor_validate_input_tensors)
+def as_tensor(
```

Comment:
@yieldthought thanks for the original implementation. We're integrating it into `ttnn` so other folks can get the functionality out of the box!